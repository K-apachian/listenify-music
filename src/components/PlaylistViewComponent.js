import axios from "axios";
import React, { useState, useEffect } from "react";
import { Container, Row, Col, Button, Accordion } from "react-bootstrap";
import { Link, useHistory, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import "../mystyles/PlaylistViewComponent.css";
import { getAllPlaylists, getSongs } from "../redux";
import usePlaylistData from "../custominfo/usePlaylistData";
import DetailSongs from "./DetailSongs";
import styled, { keyframes } from "styled-components";
import {
  pulse,
  tada,
  lightSpeedIn,
  rollIn,
  bounce,
  bounceIn,
  rubberBand,
  fadeIn,
  flash,
  zoomInUp,
  slideInUp,
  jello,
  swing,
} from "react-animations";

const AnimatePlaylistView = styled.div`
  animation: 1.5s ${keyframes`${fadeIn}`};
`;

const PlaylistViewComponent = withRouter(
  ({
    history,
    match,
    songsData,
    userData,
    playlistData,
    getAllPlaylists,
    ...props
  }) => {
    const songsObj = props.location.state || {};
    console.log("PLAYLIST VIEW", songsObj);

    const [playlist, songsInPlaylist] = usePlaylistData(
      match.params.id,
      songsObj
    );

    const playlistSongs = useHistory();
    const handlePlaylistData = (gotopath) => {
      playlistSongs.push({
        pathname: `/playlists/${match.params.id}/${gotopath}`,
        state: [playlist, songsInPlaylist],
      });
    };

    return (
      <div className="playlistview">
        <AnimatePlaylistView>
          <Container fluid className="playlistviewcontainer">
            <h2
              style={{
                color: "white",
                textAlign: "center",
              }}
            >
              <b>Welcome to {playlist.title} Playlist</b>
            </h2>
            <Row>
              <ul className="editbuttonss">
                <li>
                  <Link to="/dashboard">
                    <Button variant="info">Back to Playlist</Button>
                  </Link>
                </li>
                <li>
                  <Button
                    variant="success"
                    onClick={() => {
                      handlePlaylistData("addsongtoplay");
                    }}
                  >
                    Add Songs
                  </Button>
                </li>
                <li>
                  <Button
                    variant="danger"
                    onClick={() => {
                      handlePlaylistData("deletesongplaylist");
                    }}
                  >
                    Delete Songs
                  </Button>
                </li>
              </ul>
            </Row>
            <div className="viewaccordion scrollbar scrollbar-deep-blue">
              <Container>
                <Row>
                  <Col>
                    {songsInPlaylist.length === 0 ? (
                      <h2
                        style={{
                          color: "white",
                          textAlign: "center",
                          paddingBottom: "20px",
                        }}
                      >
                        Sorry, No Songs Found In Your Playlist !!
                      </h2>
                    ) : (
                      <Accordion
                        style={{
                          margin: "auto",
                          width: "auto",
                          alignContent: "center",
                        }}
                        defaultActiveKey=""
                      >
                        {songsInPlaylist.map((song) => (
                          <Accordion.Item eventKey={song.id} key={song.id}>
                            <Accordion.Header>
                              <span className="me-auto">
                                <b>{song.title}</b>
                              </span>
                              {/* <span className="right">
                                <b>{song.songlength}</b>
                              </span> */}
                            </Accordion.Header>
                            <Accordion.Body style={{textAlign:"center"}}>
                              <DetailSongs
                                songData={song}
                                userLogStatus={userData.isLoggedIn}
                              />
                            </Accordion.Body>
                          </Accordion.Item>
                        ))}
                      </Accordion>
                    )}
                  </Col>
                </Row>
              </Container>
            </div>
          </Container>
        </AnimatePlaylistView>
      </div>
    );
  }
);

const mapStateToProps = (state) => {
  return {
    userData: state.user,
    songsData: state.song,
    playlistData: state.playlist,
  };
};
export default connect(mapStateToProps, { getAllPlaylists })(
  PlaylistViewComponent
);
