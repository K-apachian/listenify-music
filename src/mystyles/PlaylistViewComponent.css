.playlistview {
  background-image: url("../images/playlistsongs.jpg");
  width: auto;
  height: 100vh;
  background-size: cover;
  background-position: center;
  background-attachment: scroll;
  color: #f5f5f5;
}

.playlistviewcontainer {
  padding-top: 35px;
}

.editbuttonss {
  list-style-type: none;
  justify-content: center;
  text-indent: 1cm;
  display: inline-flex;
}

@media only screen and (max-width: 600px) {
  .editbuttonss {
    list-style-type: none;
    justify-content: center;
    line-height: 1.2cm;
    text-align: center;
    display: inline-table;
  }
}
