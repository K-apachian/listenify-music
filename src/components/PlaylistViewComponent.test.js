import React from "react";
import { shallow } from "enzyme";
import { expect, it } from "@jest/globals";

import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import PlaylistViewComponent from "./PlaylistViewComponent";

const mockStore = configureMockStore();
const store = mockStore({
  userData: {
    loading: false,
    users: [],
    user: {},
    error: "",
    isLoggedIn: false,
  },
  songsData: {
    loading: false,
    songs: [],
    error: "",
  },
  playlistData: {
    loading: false,
    playlists: [],
    error: "",
  },
});

it("should render entire PlaylistViewComponent Component", () => {
  const wrapper = shallow(
    <Provider store={store}>
      <PlaylistViewComponent />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
