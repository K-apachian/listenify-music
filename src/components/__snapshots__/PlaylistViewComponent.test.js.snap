// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render entire PlaylistViewComponent Component 1`] = `ShallowWrapper {}`;
